### PR TITLE
Newmxemf

### PR DESCRIPTION
**Pull request by StoddardOXC**
rework MXE makefile to produce debug symbols, enable LTO by default

### CODE REVIEW

**Comment by MeridianOXC** [comment]
Should I replace YAML_SRC with YAML_DIR in comments and examples too?

File: src/Makefile.mxe [hunk 1]
```diff
@@ -1,73 +1,84 @@
...
+# Example:
 #
-# With mxe-supplied yaml-cpp:
+# make -f Makefile.mxe MXE_DIR=/opt/mxe ARCH=x86_64-w64-mingw32.static
 #
-# make -f Makefile.mxe ARCH=x86_64-w64-mingw32.static MXE_YAML=1
+# With local yaml-cpp:
 #
-# You can also set BIN=whatever.exe to rename the .exe
+# make -f Makefile.mxe ARCH=x86_64-w64-mingw32.static YAML_SRC=/whatever/yaml-cpp/
```
Comment (MeridianOXC) [inline src/Makefile.mxe]: YAML_DIR ?

File: src/Makefile.mxe [hunk 2]
```diff
@@ -1,73 +1,84 @@
...
 BINDIR = ../bin/
-YAML_INC ?= ../../yaml-cpp/include/
-YAML_SRC ?= ../../yaml-cpp/src/
-
 SRCS = $(wildcard *.cpp */*.cpp */*/*.cpp)
 
-# if MXE_YAML is set, use the mxe-supplied one
-ifeq ($(strip $(MXE_YAML)),)
-	SRCS += $(wildcard $(YAML_SRC)*.cpp $(YAML_SRC)*/*.cpp)
+# if YAML_SRC is set, use the supplied yaml-cpp, else MXE yaml-cpp
```
Comment (MeridianOXC) [inline src/Makefile.mxe]: YAML_DIR ?

Comment (StoddardOXC) [reply src/Makefile.mxe]: yeah, duh

let me resubmit this

**Comment by MeridianOXC** [comment]
Cherry-picked.

**Pull request closed without merging**